packages/devtools-network-console/src/ui/ResponseViewer/index.tsx

```diff
@@ -148,11 +148,14 @@ export function ResponseViewer(props: IResponseViewerProps) {
     const tabsToDisplay = PIVOT_DEFAULT_ITEMS.slice();
     if (!!renderedPreview) {
         tabsToDisplay.unshift(PIVOT_PREVIEW_ITEM);
+    } else if(currentTab === "preview") {
+        setCurrentTab('body');
     }
 
-    if (true || false) {
-        // TODO: Mock up the websocket connection
+    if (props.response.isWebsocketUpgrade) {
         tabsToDisplay.push(WEBSOCKET_ITEM);
+    } else if (currentTab === "websocket") {
+        setCurrentTab('body');
```

Comment:
I wasn't sure how best to handle this. Basically, if it's possible to have the same request_id go to different urls, you can end up in a bad state if the new request doesn't have websockets (or a preview) if that was the tab you were on previously. It happens all the time with the webhost, because we use the same id everywhere. not sure if that happens with a real host

Reply:
added comment to check if we can remove later